scripts/content.js

```diff
@@ -1,61 +1,125 @@
+ * @param  {object} observer  observer properties.
+ */
+function translateComments(mutationsList, observer) {
+  for (const mutation of mutationsList) {
+    if (mutation.type == "childList") {
+      mutation.addedNodes.forEach((node) => {
+        if (node.nodeName.toLowerCase() == "ytd-comment-renderer") {
+          let bodyElement = node.querySelectorAll(
+            "#body #main #comment-content #expander #content #content-text .style-scope.yt-formatted-string"
+          );
+
+          if (bodyElement) {
+            bodyElement.forEach((element) => {
+              const spellcheckValue = element.getAttribute("spellcheck");
+              if (
+                element.textContent.trim() != "" &&
+                spellcheckValue != "false" &&
+                element.textContent != " " &&
+                !element.textContent.match(regexEmoji)
+              ) {
+                fetch(detectURL, {
+                  method: "POST",
+                  headers: {
+                    "Content-Type": "application/json",
+                  },
+                  body: JSON.stringify({
+                    q: element.textContent,
+                  }),
+                })
+                  .then((response) => response.json())
+                  .then((data) => {
+                    language = data.data.detections[0][0].language;
+                    if (language != "en") {
+                      callTranslateAPI(element)
+                        .then((translatedText) => {
+                          if (translatedText) {
+                            element.textContent =
+                              translatedText + " (translated)";
+                            console.log("Translated text:", translatedText);
+                          }
+                        })
+                        .catch((error) => {});
+                    }
+                  })
+                  .catch((error) => {
+                    console.log("Error: ", error);
+                  });
+              }
+            });
+          }
+        }
+      });
+    }
+  }
+}
+
+/**
+ * Translates comments on click
+ */
 commentWrapper.addEventListener("click", (event) => {
   const comment = event.target.closest("yt-formatted-string");
-  if (comment) {
-    fetch(url, {
-      method: "POST",
-      headers: {
-        "Content-Type": "application/json",
-      },
-      body: JSON.stringify({
-        q: comment.textContent,
-        target: "en",
-      }),
-    })
-    .then(response => response.json())
-    .then(data => {
-      const translatedComment = data.data.translations[0].translatedText;
-      console.log(translatedComment);
-      comment.textContent = translatedComment;
-    })
-    .catch(error => {
-      console.log("Error: ", error);
+
+  callTranslateAPI(comment)
+    .then((translatedText) => {
+      if (translatedText) {
+        comment.textContent = translatedText;
+        console.log("Translated text:", translatedText);
```

Comment:
doesnt this still have print to console twice with line 84?

Reply:
this will console log on click vs the other console logs on load, thought we were separating the functionalities